src/zenml/cli/stack.py

```diff
@@ -447,13 +465,21 @@ def update_stack(
     help="Share a stack and all its components.",
 )
 @click.argument("stack_name_or_id", type=str, required=False)
+@click.option(
+    "--recursive",
+    "-r",
+    "recursive",
+    is_flag=True,
+    help="Recursively also share all stack components if they are private.",
+)
 def share_stack(
-    stack_name_or_id: Optional[str],
+    stack_name_or_id: Optional[str], recursive: bool = False
```

Comment:
This changes the default behavior from sharing to not sharing, is that intended?